### PR TITLE
fix: bump min cdk version to 2.28.0 to use node 14

### PR DESCRIPTION
**Pull request by pahud**
- As node12 runtime has been deprecated, the integration test using aws-cdk-lib@2.0.0 synthesizes the custom resource provider with runtime node12 which fails the deployment. We should use nodejs14 instead. And the minimal CDK release for node14 custom resource runtime is [2.28.0](https://github.com/aws/aws-cdk/releases/tag/v2.28.0).
- update contributing guide to execute the command with `npx projen` to avoid confusion.
- generate `.gitpod.yml` from projen to use the jsii/superchain container image that comes with everything required to run the `yarn build`.

### CODE REVIEW

File: .projenrc.ts [hunk 1]
```diff
@@ -11,7 +11,7 @@ const project = new awscdk.AwsCdkConstructLibrary({
   projenrcTs: true,
   author: 'Amazon Web Services',
   authorAddress: 'aws-cdk-dev@amazon.com',
-  cdkVersion: '2.0.0',
+  cdkVersion: '2.83.0',
```
Comment (kaizencc) [inline .projenrc.ts]: same comment about the cdkVersion being the minimal compatible version. Don't think we should change

**Review by kaizencc — CHANGES_REQUESTED** [review]
Looks like some changes here leaked into #299 as well. lets isolate them here and then this PR can be backported everywhere

File: .projenrc.ts (hunk 1 above)
Comment (pahud) [reply .projenrc.ts]: Just changed to 2.28.0 as described in the PR description.

**Comment by pahud** [comment]
@kaizencc  please take a look again.

File: .projenrc.ts [hunk 2]
```diff
@@ -73,4 +73,14 @@ new WorkflowNoDockerPatch(project, { workflow: 'release', workflowName: `release
 
 project.preCompileTask.exec('layer/build.sh');
 
+// For gitpod users, use jsii/superchain as the dockerImage for the workspace.
+let gitpod = new Gitpod(project, {
```
Comment (kaizencc) [inline .projenrc.ts]:
```suggestion
const gitpod = new Gitpod(project, {
```

**Review by kaizencc — APPROVED** [review]

**Comment by mrgrain** [comment]
@mergifyio backport kubectl-v22/main

**Comment by mrgrain** [comment]
@mergifyio backport kubectl-v22/main

**Comment by mrgrain** [comment]
@Mergifyio copy kubectl-v21/main kubectl-v22/main

**Comment by mrgrain** [comment]
@mergifyio copy kubectl-v21/main kubectl-v22/main

**Pull request merged**